### PR TITLE
adds ability to nest CDN query conditions

### LINKED ISSUE #30: expand model find capabilities to find by references
API should resemble:
```ruby
Page.find_by(subpages: { id: 'asdf234' })
```

see https://www.contentful.com/r/knowledgebase/searching-on-references/

### PR DESCRIPTION
**Pull request by jpowell**
closes #30

### CODE REVIEW

**Comment by jpowell** [comment]
@gburgett still a WIP. Some things need to move around and be cleaned up / consolidated, but have the basis for moving forward.

File: lib/wcc/contentful/store/cdn_adapter.rb [hunk 1]
```diff
@@ -48,15 +48,78 @@ def initialize(client, relation)
...
+        type
+        createdAt
+        updatedAt
+        revision
+        locale
+
+        contentType
+        space
+        environment
+      ].freeze
```
[the remark this answers is not in the document]
Comment (jpowell) [reply lib/wcc/contentful/store/cdn_adapter.rb]: Likely, but that raises the question if you should?

File: lib/wcc/contentful/store/base.rb [hunk 2]
```diff
@@ -108,6 +108,14 @@ def apply(filter, context = nil)
       def op?(key)
         OPERATORS.include?(key.to_sym)
       end
+
+      def sys?(field)
+        field.to_s =~ /sys/
```
Comment (gburgett) [inline lib/wcc/contentful/store/base.rb]: What if I name my field one of the following:

* `sysadmin`
* `ecosystem`
* `systemic`
* `sysopsEmail`
* `photosynthesys` - b/c I misspelled it

Comment (jpowell) [reply lib/wcc/contentful/store/base.rb]: good call - regression test written and fixed

**Pull request merged**